phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -0,0 +1,550 @@
+            false,
+            "[Optional]If passed this will output steps that will be executed");
+    private static final Option HELP_OPTION = new Option("h", "help",
+            false, "Help");
+    private static final Option LOG_FILE_OPTION = new Option("lf", "logfile",
+            true,
+            "Log file path where the logs are written");
+    private static final Option HOST_OPTION = new Option("ho", "host",
+            true, "[Required]zookeeper quorum host");
+    private static final Option PORT_OPTION = new Option("po", "port",
+            true, "[Required]zookeeper port");
+    private static final Option COMPACTION_OPTION = new Option("c", "compaction",
+            true, "[Optional]enable/disable compaction will be handled by the tool");
+
+    public static final String UPGRADE_OP = "upgrade";
+    public static final String ROLLBACK_OP = "rollback";
+
+
+    private HashMap <String, Boolean> tableMC = new HashMap<>();
+    private HashMap<String, ArrayList<String>> tablesAndIndexes;
+    private HashMap<String, String> prop = new  HashMap<>();
+
+    private boolean dryRun, upgrade, compaction;
+    private String connectionURL;
+    private String operation;
+    private String inputTables;
+    private String logFile;
+    private String inputFile;
+
+    public void setDryRun(boolean dryRun) {
+        this.dryRun = dryRun;
+    }
+
+    public void setInputTables(String inputTables) {
+        this.inputTables = inputTables;
+    }
+
+    public void setLogFile(String logFile) {
+        this.logFile = logFile;
+    }
+
+    public void setInputFile(String inputFile) {
+        this.inputFile = inputFile;
+    }
+
+    public boolean getDryRun() {
+        return this.dryRun;
+    }
+
+    public String getInputTables() {
+        return this.inputTables;
+    }
+
+    public String getLogFile() {
+        return this.logFile;
+    }
+
+    public String getOperation() {
+        return operation;
+    }
+
+    public String getConnectionURL() {
+        return connectionURL;
+    }
+
+    public boolean getCompaction() {
+        return compaction;
+    }
+
+    public static void main (String[] args) {
+        CommandLine cmdLine = null;
+
+        IndexUpgradeTool iut = new IndexUpgradeTool();
+        try {
+            cmdLine = iut.parseOptions(args);
+            LOGGER.info("Index Upgrade tool initiated: "+ String.join(",", args));
+        } catch (IllegalStateException e) {
+            iut.printHelpAndExit(e.getMessage(), iut.getOptions());
+        }
+        iut.initializeTool(cmdLine);
+        iut.prepareToolSetup();
+        iut.executeTool();
+    }
+
+    public IndexUpgradeTool(String mode, String tables, String inputFile, String host, String port,
+            String outputFile, boolean dryRun, boolean compaction) {
+        this.operation = mode;
+        this.inputTables = tables;
+        this.inputFile = inputFile;
+        this.connectionURL = "jdbc:phoenix:" + host + ":" + port;
```

Comment:
any reason we can't use ZK quorum and ConnectionUtil.getInputConnection? 
> Priyank also mentioned this initially. When I tried using it, IT was failing and now I know why! This tool is like a client executing necessary commands similar to sqlline, didn't realize tool is extending configured class which I can make use of and get rid of host/port as well. ll address this. 
Will this be sufficient to talk to a secure cluster? (Where would the user/principal be set?)
> This would change as I use getInputConnection method. I don't have to rely on URL anymore.